app/models/mixins/cloud_tenancy_mixin.rb

```diff
@@ -0,0 +1,19 @@
+module CloudTenancyMixin
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    include TenancyCommonMixin
+
+    def tenant_id_clause_format(tenant_ids)
+      ["(tenants.id IN (?) AND ext_management_systems.tenant_mapping_enabled IS TRUE) OR ext_management_systems.tenant_mapping_enabled IS FALSE", tenant_ids]
+    end
+
+    def tenant_joins_clause(scope)
```

Comment:
I found the issue. In this PR the joins_clause uses joins where as the second PR in the cloud tenant series, https://github.com/ManageIQ/manageiq/pull/14036/commits/e471dbbc20e3992aab5948cf27feb5b0f3fde084, switched over to using includes. I switched the second patch to using includes, because some tests were failing and I had thought it was an issue with missing attributes in the test data. But I now understand that if the cloud_tenant_mapping is not enabled, then an inner join to the cloud_tenants table would always return 0 results, so an outer join (includes) is actually necessary. As far as I can tell, the outer join is only required against cloud_tenant and not ext_management_system.

I will revise this PR and rebase the second in this series.